File: auriga_test/public/js/item.js
frappe.ui.form.on("Item", {
	onload : function(frm){
		if (frm.doc.__islocal){
			auto_populate_fields(frm)
		}
	},
});

// while creating single varient below fields will be autopopulate from attributes
function auto_populate_fields(frm){
	var childTableRows = cur_frm.doc.attributes || [];
	for (var i = 0; i < childTableRows.length; i++) {
		var row = childTableRows[i];
		if (row.attribute == 'Height'){
			cur_frm.doc.height = row.attribute_value
		}
		if (row.attribute == 'Width'){
			cur_frm.doc.width = row.attribute_value
		}
		if (row.attribute == 'Yield'){
			cur_frm.doc.yield_ = row.attribute_value
		}
      	frm.refresh_fields();
    }
}

File: auriga_test/auriga_test/docevents/item.py
import frappe

# this will auto set custom fields value, and it will be useful when doing multiple variants item, as they are created through backend
def update_conversion_factor(doc,event):
	if doc.attributes and doc.variant_of:
		for row in doc.attributes:
			if row.get('attribute') == 'Height':
				doc.height = int(row.get('attribute_value'))
			if row.get('attribute') == 'Width':
				doc.width = int(row.get('attribute_value'))
			if row.get('attribute') == 'Yield':
				doc.yield_ = float(row.get('attribute_value'))
	if not doc.has_variants:
		update_uoms_conversion(doc)

# Update UOM Conversion table
def update_uoms_conversion(doc):
	for row in doc.uoms:
		if row.formula:
			row.conversion_factor = eval(str(row.formula))
